File: Algorithms/p153_Find_Minimum_in_Rotated_Sorted_Array/p153_Find_Minimum_in_Rotated_Sorted_Array.py
class Solution(object):
    def findMin(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        l = 0;
        r = len(nums)-1
        while l<r:
            mid = (l+r)//2
            x = nums[mid]
            if nums[l]>x:
                r = mid
            elif x>nums[r]:
                l = mid+1
            else:
                break
        return nums[l]

File: Algorithms/p153_Find_Minimum_in_Rotated_Sorted_Array/p153_Find_Minimum_in_Rotated_Sorted_Array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int l = 0;
        int r = nums.size()-1;
        while(l<r){
            int mid = (l+r)/2;
            int x = nums[mid];
            if(nums[l]>x){
                r = mid;
            }else if(x>nums[r]){
                l = mid+1;
            }else{
                break;
            }
        }
        return nums[l];
    }
};
